cli/tests/extensions.rs

```diff
@@ -280,6 +280,135 @@ fn conflicting_extension_name_is_filtered() {
     assert!(output.contains("extension was filtered out"));
 }
 
+mod permissions {
+    use std::ffi::OsStr;
+
+    use assert_cmd::assert::Assert;
+    use predicates::prelude::*;
+
+    use super::*;
+
+    struct TestCLI {
+        tempdir: TempDir,
+        cwd: Option<PathBuf>,
+    }
+
+    impl Default for TestCLI {
+        fn default() -> Self {
+            Self { tempdir: TempDir::new().unwrap(), cwd: None }
+        }
+    }
+
+    impl TestCLI {
+        fn new() -> Self {
+            Default::default()
+        }
+
+        fn cwd(mut self, cwd: PathBuf) -> Self {
+            self.cwd = Some(cwd);
+            self
+        }
+
+        fn install_extension(&self, path: &Path) -> Assert {
+            self.run(&["extension", "install", "-y", &path.to_string_lossy()])
+        }
+
+        fn run<S: AsRef<str> + AsRef<OsStr>>(&self, args: &[S]) -> Assert {
+            let mut cmd = Command::cargo_bin("phylum").unwrap();
+
+            cmd.env("XDG_DATA_HOME", self.tempdir.path()).args(args);
+
+            if let Some(cwd) = self.cwd.as_ref() {
+                cmd.current_dir(cwd);
+            }
+
+            cmd.assert()
+        }
```

Comment:
This is exactly the kinda stuff I'd love to share between normal and end-to-end integration tests. I feel like we'll probably need this a bunch anyway.